src/server/pfs/server/server_commit.go

```diff
@@ -497,6 +483,18 @@ func (a *apiServer) resolveCommit(ctx context.Context, sqlTx *pachsql.Tx, commit
 	if commitHandle.Id == "" && commitHandle.GetBranch().GetName() == "" {
 		return nil, errors.Errorf("cannot resolve commit with no ID or branch")
 	}
+	if commitHandle.AccessRepo().Name == fileSetsRepo {
+		cinfo := &pfs.CommitInfo{
+			Commit:      commitHandle,
+			Description: "FileSet - Virtual Commit",
+			Finished:    &timestamppb.Timestamp{}, // it's always been finished. How did you get the id if it wasn't finished?
+		}
+		return &pfsdb.Commit{
+			ID:         0, // this doesn't seem like the right thing to do, but here we are.
+			CommitInfo: cinfo,
+			Revision:   0,
+		}, nil
+	}
```

Comment:
Can you explain what's going on with the green block here ;) ?

Reply:
Yeah I copy pasted this from another function. There's some thing where filesets get put in a repo. I don't know why we do this. Maybe its an optimization to make compaction easier to do. These commits aren't real though. They don't exist in the database. Again I have no idea why. I know this is a terrible answer, but yeah 🤷‍♂️